src/main/java/edu/cornell/kfs/krad/service/impl/ClamAVAntiVirusServiceImpl.java

```diff
@@ -0,0 +1,217 @@
+                DataOutputStream socketDataOutput = new DataOutputStream(socketOutput);
+            ) {
+                return operation.run(socketReader, socketDataOutput);
+            }
+        } catch (Exception e) {
+            LOG.error("performClamAVOperation, Exception occurred while running ClamAV operation", e);
+            return errorResultBuilder.apply(e);
+        }
+    }
+
+    public String stats() {
+        return runSimpleCommand(ClamAVCommands.STATS);
+    }
+
+    public boolean ping() {
+        String pingResponse = runSimpleCommand(ClamAVCommands.PING);
+        return StringUtils.equals(ClamAVResponses.PING_RESPONSE_OK, pingResponse);
+    }
+
+    private String runSimpleCommand(String command) {
+        return performClamAVOperation(
+                (socketReader, socketOutput) -> runSimpleCommand(command, socketReader, socketOutput),
+                exception -> KFSConstants.EMPTY_STRING);
+    }
+
+    private String runSimpleCommand(String command, Reader socketReader, DataOutputStream socketOutput)
+            throws IOException {
+        sendSimpleCommandToClamAV(command, socketOutput);
+        socketOutput.flush();
+        return readClamAVResponse(socketReader);
+    }
+
+    private void sendSimpleCommandToClamAV(String command, DataOutputStream socketOutput) throws IOException {
+        if (StringUtils.isBlank(command)) {
+            throw new IllegalArgumentException("command cannot be blank");
+        }
+        String delimitedCommand = ClamAVDelimiters.Z_PREFIX + command + ClamAVDelimiters.NULL_SUFFIX;
+        byte[] delimitedCommandAsBytes = delimitedCommand.getBytes(StandardCharsets.UTF_8);
+        socketOutput.write(delimitedCommandAsBytes);
+    }
+
+    private String readClamAVResponse(Reader socketReader) throws IOException {
+        char[] responseChunk = new char[DEFAULT_CHUNK_SIZE];
+        StringBuilder fullResponse = new StringBuilder(DEFAULT_CHUNK_SIZE);
+        int responseTerminatorIndex = -1;
+        int chunkLength;
+        
+        do {
+            chunkLength = socketReader.read(responseChunk);
+            if (chunkLength > 0) {
+                fullResponse.append(responseChunk, 0, chunkLength);
+                responseTerminatorIndex = fullResponse.indexOf(
+                        ClamAVDelimiters.NULL_SUFFIX, fullResponse.length() - chunkLength);
+            }
+        } while (responseTerminatorIndex == -1 && chunkLength > 0);
+        
+        if (responseTerminatorIndex == -1) {
+            LOG.error("readClamAVResponse, End of stream was reached before reading full ClamAV response!  " +
+                    "No content will be returned.  " + fullResponse.length() +
+                    " characters were received prior to abrupt end.");
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("readClamAVResponse, Response text received prior to stream end: " + fullResponse);
```

Comment:
should this be a debug log.  Would this be helpful in all cases?

Reply:
Under further consideration, I agree that it makes sense to have this content visible in all cases and just debug-logging ones. I will update this logging accordingly.